pr_agent/algo/pr_processing.py

```diff
@@ -24,10 +24,10 @@ def get_pr_diff(git_provider: Union[GithubProvider, Any], token_handler: TokenHa
     Returns a string with the diff of the PR.
     If needed, apply diff minimization techniques to reduce the number of tokens
     """
-    files = list(git_provider.get_diff_files())
+    git_provider.pr.files = list(git_provider.get_diff_files())
```

Comment:
Consider using a getter method to access the 'files' attribute of 'pr' object. Directly accessing or modifying object attributes is not a good practice as it breaks the encapsulation principle of OOP. [important]